lib/good_migrations/configuration.rb

```diff
@@ -0,0 +1,29 @@
+module GoodMigrations
+  class Configuration
+    class << self
+      # Migrations with timestamps (the numbers at the beginning of the file name) from
+      # before this configured time will be allowed to perform autoloading, bypassing the
+      # mechanism of this gem. Accepts:
+      #   nil: meaning never permit it
+      #   String accepted by `Time.parse`, such as: 20211103150610 or 20211103_150610
+      #   object responding to `to_time`, such as Date and Time
+      def permit_autoloading_before_date=(value)
+        case value
+        when nil
+          # Stay nil
+        when String
+          value = Time.parse(value)
+        else
+          if value.respond_to?(:to_time)
+            value = value.to_time
+          else
+            raise "Received an invalid value for permit_autoloading_before_date: #{value.inspect}"
+          end
+        end
+        @permit_autoloading_before_date = value
```

Comment:
This might look more natural to me:

```suggestion
        return unless value.present?
        
        @permit_autoloading_before_date = if value.is_a?(String)
          Time.parse(value)
        elsif value.respond_to?(:to_time)
          value.to_time
        else
          raise "Received an invalid value for permit_autoloading_before_date: #{value.inspect}"
        end
```

Reply:
I agree I didn't like the look of that code. However, your proposed change would do nothing on `nil` instead of setting `@permit_autoloading_before_date` to `nil`

Reply:
I came up with this:
```
if value.is_a?(String)
  value = Time.parse(value)
elsif value.respond_to?(:to_time)
  value = value.to_time
elsif !value.nil?
  raise "Received an invalid value for permit_autoloading_before_date: #{value.inspect}"
end
@permit_autoloading_before_date = value
```

I'll make the switch if you prefer that